File: pancakes.js
// Usage: cat input | node pancakes.js

const happy = "+"
const blank = "-"

function getFlipCount(input) {
    height = substr_count(input, happy + blank) + substr_count(input, blank + happy) + 1;
    if (input.substr(-1) == blank) {
        return height;
    }
    return height - 1;
}

function substr_count(string, substring) {
    let n = 0;
    let pos = 0;
    let step = substring.length;

    while (true) {
        pos = string.indexOf(substring, pos);
        if (pos >= 0) {
            ++n;
            pos += step;
        } else break;
    }
    return n;
}

function handleInput() {
    const readline = require('readline');

    const rl = readline.createInterface({
        input: process.stdin,
        terminal: false,
    });

    let i = 0;
    let count;
    rl.on('line', line => {
        // Store number of expected input lines
        if (i == 0) {
            count = parseInt(line.trim());
            i++;
            return;
        }

        // Stop reading input after expected number of lines
        if (i > count) {
            return;
        }

        let input = line.trim();
        console.log(`Case #${i}: ${getFlipCount(input)}`);

        i++;
    });
}

handleInput();

File: pancakes.py
# Usage: cat input | python pancakes.py
import sys

happy = '+'
blank = '-'

def getFlipCount(input):
    height = input.count(happy + blank) + input.count(blank + happy) + 1
    if input.endswith(blank):
        return height
    return height - 1

def main():
    count = None
    for i, line in enumerate(sys.stdin):
        if i == 0:
            count = int(line.strip())
        if i > count:
            break

        print("Case #%d: %d" % (i, getFlipCount(line.strip())))

main()
